pymc3_hmm/distributions.py

```diff
@@ -186,7 +187,9 @@ def logp(self, obs):
 
         obs_tt = tt.as_tensor_variable(obs)
 
-        logp_val = tt.alloc(-np.inf, *obs.shape)
+        shape_var = tuple(obs_tt.shape.tag.test_value)
+
+        logp_val = tt.alloc(-np.inf, *shape_var)
```

Comment:
We can't add the shared variables here.  Instead, we will need to rewrite a log-likelihood graph returned by this method.